dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1141,6 +1158,52 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    HKEY hVerbKey;
+    WCHAR VerbKey[sizeof("shell\\") + MAX_VERB];
+    hr = StringCbPrintfW(VerbKey, sizeof(VerbKey), L"shell\\%s", pEntry->Verb.GetString());
+    if (SUCCEEDED(hr) && m_pDataObj &&
+        RegOpenKeyExW(pEntry->hkClass, VerbKey, 0, KEY_READ, &hVerbKey) == ERROR_SUCCESS)
+    {
+        CLSID clsid;
+        BOOL Done = FALSE;
+
+        DWORD KeyState = 0;
+        if (lpcmi->fMask & CMIC_MASK_SHIFT_DOWN)
+            KeyState |= MK_SHIFT;
+        if (lpcmi->fMask & CMIC_MASK_CONTROL_DOWN)
+            KeyState |= MK_CONTROL;
+
+        POINTL *pPtl = NULL;
+        C_ASSERT(sizeof(POINT) == sizeof(POINTL));
+        if (lpcmi->fMask & CMIC_MASK_PTINVOKE)
+            pPtl = (POINTL*)&lpcmi->ptInvoke;
+
+        if (!Done)
+        {
+            // TODO: IExecuteCommand
+        }
+
+        if (!Done)
```

Comment:
Here, I can understand you may want to keep that one here, in case the to-do IExecuteCommand above will modify `Done`, in the future.